moveit_core/planning_scene/include/moveit/planning_scene/planning_scene.h

```diff
@@ -678,7 +679,7 @@ class PlanningScene : private boost::noncopyable, public std::enable_shared_from
   double distanceToCollisionUnpadded(const robot_state::RobotState& robot_state,
                                      const collision_detection::AllowedCollisionMatrix& acm) const
   {
-    return getCollisionWorld()->distanceRobot(*getCollisionRobotUnpadded(), robot_state, acm);
+    return getCollisionRobotUnpadded()->distanceRobot(robot_state, acm);
```

Comment:
```suggestion
    return getCollisionEnvUnpadded()->distanceRobot(robot_state, acm);
```

Reply:
I have kept the getCollisionRobot[Unpadded] / getCollisionWorld functions for now and made them return the same respective collision environment. As these function might be used also in user code, I would add an ROS_ERROR_MESSAGE stating that these functions will be eventually deprecated and getCollisionEnv[Unpadded] should be used. Or what is the common way to handle functions which will be deprecated in the future?

But throughout the moveit codebase, I should change all getCollisionWorld / getCollisionRobot to a new getCollisionEnv function right?

Reply:
IIRC the currently correct way is to mark it deprecated using the moveit macros for one release or year (?), and then remove it. See the discussion and link to the macros [here](https://github.com/ros-planning/moveit/issues/1229)

Reply:
That's fair. Not sure if it's considered marking deprecated if we're changing the return type, but I guess this would still work okay.

Reply:
You're right Bryce we are changing the return type so users will get errors anyway...